### ISSUE TITLE
False negative MultilineLambdaItParameter on complex multiline single statement

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#5393]
## Expected Behavior
Expectation to pass this test:
```kotlin
    @Nested
    inner class `single parameter, multiline lambda with a single statement` {

        @Test
        fun `reports when statement is spanning multiple lines`() {
            val code = """
            fun f() {
                val digits = 1234.let {
                    check(it > 0) {
                        // This can be arbitrarily long or complex.
                        "${'$'}it must be positive."
                    }
                }
            }
            """.trimIndent()
            val findings = subject.compileAndLintWithContext(env, code)
            assertThat(findings).hasSize(1)
        }
```

## Observed Behavior
```kotlin
fun Feed.verifyHasAllAttributes(attributes: Set<Feed.Attribute>) {
	attributes.forEach { // implicit `it ->` here wasn't flagged.
		assertTrue(this.attributes.contains(it)) {
			"$it not found in ${this.attributes}"
		}
	}
}

fun Feed.verifyAllAttributesAreValid() {
	this.attributes.forEach { // implicit `it ->` here was flagged.
		assertNotNull(it.code)
		assertNotNull(it.title)
	}
}
```

```
> Task :backend:feed:detektTest FAILED
style - 20min debt
        MultilineLambdaItParameter - [The implicit `it` should not be used in a multiline lambda. Consider giving your(...)] at backend\feed\src\test\kotlin\net\twisterrob\cinema\cineworld\sync\syndication\FeedData.kt:80:26
```

## Steps to Reproduce
 * Clone https://github.com/TWiStErRob/net.twisterrob.cinema
 * Check out 92bca07146165cc17693b0b6fa95be7d47baf550
 * `gradlew :backend:feed:detektTest`
 * Look for FeedData.kt

## Context
I think this should be flagged if the single statement spans more than 1 line. At the point `it` is captured in any lambda, the complexity goes way up, and an explicit name helps readability greatly same as any other flagging of `MultilineLambdaItParameter`. I think the example clearly shows that a mutli-line single-statement lambda can be very easily more complex than a multi-statement lambda.

## Your Environment

* Version of detekt used: 1.21.0
* Link to your project (if it's a public repository): https://github.com/TWiStErRob/net.twisterrob.cinema/blob/490998a42ecda26515ff761db6f51a224130a993/Heroku/backend/feed/src/test/kotlin/net/twisterrob/cinema/cineworld/sync/syndication/FeedData.kt#L67-L84
